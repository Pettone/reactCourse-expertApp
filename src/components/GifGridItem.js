import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ( { id, title, url} ) => {

    console.log({id,title,url});

    return (
        <div className = "card animate__animated animate__bounce animate__delay-2s">
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}

GifGridItem.propTypes = {
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}
