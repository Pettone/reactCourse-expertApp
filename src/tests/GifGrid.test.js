import React from 'react';
import { render } from '@testing-library/jest-dom';
import { shallow } from "enzyme";
import { GifGrid } from "../components/GifGrid";
import { useFetchGifs } from '../hooks/useFetchGifs';
jest.mock('../hooks/useFetchGifs');

describe('Testing GifGrid component', () => { 

    const category = 'Category';

    test('Must to show itself correctly', () => { 

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow( <GifGrid category={ category } /> );
        expect(wrapper).toMatchSnapshot();
    });

    test('Must to show items when images are loaded by useFetchGifs', () => { 

        const gifs = [{
            id: 'ABC',
            url: 'https://localhost:8080/image.jpg',
            title: 'image title'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        const wrapper = shallow( <GifGrid category={ category } /> );
        
        expect( wrapper.find('p').exists() ).toBe(false);
        expect( wrapper.find('GifGridItem').length ).toBe(gifs.length);
    });
});