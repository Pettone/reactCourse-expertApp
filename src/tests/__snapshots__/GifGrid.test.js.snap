// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifGrid component Must to show itself correctly 1`] = `
<Fragment>
  <h3>
    Category
  </h3>
  <p>
    Loading
  </p>
  <div
    className="card-grid"
  >
    <ol />
  </div>
</Fragment>
`;
